p8e-util/src/main/kotlin/io/p8e/crypto/SmartKeySigner.kt

```diff
@@ -43,27 +71,39 @@ class SmartKeySigner(
             .data(data)
             .deterministicSignature(true)
 
-        val signatureResponse = SignAndVerifyApi().sign(sObjUUID, signatureRequest)
+        val signatureResponse = SignAndVerifyApi().sign(keyUuid, signatureRequest)
 
         return Signature.newBuilder()
             .setAlgo(DigestAlgorithm.SHA512.value)
             .setProvider("SmartKey")
             .setSignature(signatureResponse.signature.toString())
-            .setKeyId(signatureResponse.kid.toProtoUuidProv())
             .build()
-            .takeIf { verify(data, it) }
+            .takeIf { verify(data, signatureResponse.signature) }
             .orThrow { IllegalStateException("Invalid signature") }
     }
 
+    /**
+     * Get and convert SmartKey's public key (Sun Security Provider) into a BouncyCastle Provider (P8e).
+     *
+     * @return [PublicKey] return the Java security version of the PublicKey.
+     */
+    fun getPublicKey(): PublicKey {
+        val smPublicKey = SecurityObjectsApi().getSecurityObject(keyUuid).pubKey
+        val x509PublicKey = KeyFactory.getInstance("EC").generatePublic(X509EncodedKeySpec(smPublicKey))
+        val bcPublicKey = BCECPublicKey(x509PublicKey as ECPublicKey, BouncyCastlePQCProvider.CONFIGURATION)
+        return bcPublicKey.toHex().toJavaPublicKey()
+    }
+
     private fun verify(data: String, signature: Signature): Boolean = verify(data, signature)
 
-    private fun verify(data: ByteArray, signature: Signature): Boolean {
+    private fun verify(data: ByteArray, signature: ByteArray): Boolean {
+        //TODO: Investigate local verification: https://app.clubhouse.io/figure/story/90457/investigate-signature-verification-for-smartkey
```

Comment:
Added a ticket to investigate local signature verification, tried to do it straight-up but the verification came back as false positive.